### PR TITLE
Allows the nodejs launcher to recognize that certain types of errors were printed, ensuring we don't cascade less relevant messages.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Not ready for review.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@swgillespie We could go with this approach (where we have a magic string constant to return). Or, if you are willing, we could regen the proto stuff on your machine based on the change I made.  Let me know what you would prefer!

**Comment by swgillespie** [comment]
> @swgillespie We could go with this approach (where we have a magic string constant to return). Or, if you are willing, we could regen the proto stuff on your machine based on the change I made. Let me know what you would prefer!

I would prefer the latter; we have enough magic strings as it is IMO.

File: sdk/nodejs/cmd/pulumi-language-nodejs/main.go [hunk 1]
```diff
@@ -367,9 +367,16 @@ func (host *nodeLanguageHost) Run(ctx context.Context, req *pulumirpc.RunRequest
...
 		if exiterr, ok := err.(*exec.ExitError); ok {
-			// If the program ran, but exited with a non-zero error code.  This will happen often, since user
-			// errors will trigger this.  So, the error message should look as nice as possible.
+			// If the program ran, but exited with a non-zero error code.  This will happen often,
+			// since user errors will trigger this.  So, the error message should look as nice as
+			// possible.
 			if status, stok := exiterr.Sys().(syscall.WaitStatus); stok {
+				// 32 is the special exit code that means "we already logged all messages with the
+				// user, no need to report anything else.
+				if status.ExitStatus() == 32 {
```
Comment (swgillespie) [inline sdk/nodejs/cmd/pulumi-language-nodejs/main.go]: can we make this a constant so it's clear that 32 has some semantics here?

Comment (CyrusNajmabadi) [reply sdk/nodejs/cmd/pulumi-language-nodejs/main.go]: sure :)

**Pull request merged**